src/devices/acurite.c

```diff
@@ -996,6 +996,13 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
                 valid++;
             }
             else if (message_type == ACURITE_MSGTYPE_RAINFALL) {
+                // If the row is missing zeros, it can still pass the checksum.
+                // The earlier length check passes as ACURITE_TXR_BITLEN is 56.
+                if (browlen < 8) {
+                    decoder_log(decoder, 2, __func__, "skipping wrong len");
+                    continue;
+                } // DECODE_ABORT_LENGTH
```

Comment:
I noticed that this causes a zero return, and that is marked as legacy. I could refactor this function to the new return codes, but I think it makes more sense to just fix the bug first. Let me know if that should instead be done in this PR.

Reply:
Looks good. Clean up of return codes in a new PR would be great, if you have the time!